tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -259,23 +347,44 @@ internal static void SetTagObject(Span span, string key, object? value)
                     }
 
                     break;
-                case uint ui: // TODO: Can't get here from OTEL API, test with Activity API
+                case uint ui:
                     span.SetMetric(key, ui);
                     break;
-                case long l: // TODO: Can't get here from OTEL API, test with Activity API
+                case long l:
                     span.SetMetric(key, l);
                     break;
-                case ulong ul: // TODO: Can't get here from OTEL API, test with Activity API
+                case ulong ul:
                     span.SetMetric(key, ul);
                     break;
-                case float f: // TODO: Can't get here from OTEL API, test with Activity API
+                case float f:
                     span.SetMetric(key, f);
                     break;
                 case double d:
                     span.SetMetric(key, d);
                     break;
                 case IEnumerable enumerable:
-                    AgentSetOtlpTag(span, key, JsonConvert.SerializeObject(enumerable));
+                    if (allowUnrolling)
+                    {
+                        var index = 0;
+                        foreach (var element in (enumerable))
+                        {
+                            // we are only supporting a single level of unrolling
+                            SetTagObject(span, $"{key}.{index}", element, allowUnrolling: false);
+                            index++;
+                        }
+
+                        if (index == 0)
+                        {
+                            // indicates that it was an empty array, we need to add the tag
+                            AgentSetOtlpTag(span, key, JsonConvert.SerializeObject(value));
```

Comment:
If this is an empty array, then we know that `JsonConvert.SerializeObject` will return `"[]"` right? Can we just use that directly?Even if it's multi-dimensional I think it serializes the same? 🤔 

```suggestion
                            // indicates that it was an empty array, we need to add the tag
                            AgentSetOtlpTag(span, key, "[]");
```